spark/src/main/java/org/apache/iceberg/actions/BaseAction.java

```diff
@@ -86,17 +90,22 @@ protected String metadataTableName(MetadataTableType type) {
   }
 
   protected Dataset<Row> buildValidDataFileDF(SparkSession spark) {
-    String allDataFilesMetadataTable = metadataTableName(MetadataTableType.ALL_DATA_FILES);
+    return buildValidDataFileDF(spark, table().toString());
+  }
+
+  protected Dataset<Row> buildValidDataFileDF(SparkSession spark, String tableName) {
+    String allDataFilesMetadataTable = metadataTableName(tableName, MetadataTableType.ALL_DATA_FILES);
     return spark.read().format("iceberg").load(allDataFilesMetadataTable).select("file_path");
   }
 
-  protected Dataset<Row> buildManifestFileDF(SparkSession spark) {
-    String allManifestsMetadataTable = metadataTableName(MetadataTableType.ALL_MANIFESTS);
+  protected Dataset<Row> buildManifestFileDF(SparkSession spark, String tableName) {
+    String allManifestsMetadataTable = metadataTableName(tableName, MetadataTableType.ALL_MANIFESTS);
     return spark.read().format("iceberg").load(allManifestsMetadataTable).selectExpr("path as file_path");
   }
 
-  protected Dataset<Row> buildManifestListDF(SparkSession spark, Table table) {
-    List<String> manifestLists = getManifestListPaths(table);
+  protected Dataset<Row> buildManifestListDF(SparkSession spark, String tableName, TableOperations ops) {
+    Table snapshot = new BaseTable(ops, tableName);
+    List<String> manifestLists = getManifestListPaths(snapshot);
```

Comment:
What about changing `getManifestListPaths` to accept `Iterable<Snapshot>`? Then you wouldn't need to create a `BaseTable` out of a `StaticTableOperations`. Instead you could just pass `staticOps.current().snapshots()` here and `table.snapshots()` elsewhere.

Reply:
Sounds good to me